src/openms/source/ANALYSIS/OPENSWATH/MRMIonSeries.cpp

```diff
@@ -0,0 +1,566 @@
+        ionstring += "x";
+        pos = ion.getMonoWeight(Residue::XIon, tr.getProduct().getChargeState()) / (double) tr.getProduct().getChargeState();
+      }
+      else if (interpretation.hasCVTerm("MS:1001220") && interpretation.hasCVTerm("MS:1000903"))
+      {
+        ion = sequence.getSuffix(interpretation.getCVTerms()["MS:1000903"][0].getValue().toString().toInt());
+        ionstring += "y";
+        pos = ion.getMonoWeight(Residue::YIon, tr.getProduct().getChargeState()) / (double) tr.getProduct().getChargeState();
+      }
+      else if (interpretation.hasCVTerm("MS:1001230") && interpretation.hasCVTerm("MS:1000903"))
+      {
+        ion = sequence.getSuffix(interpretation.getCVTerms()["MS:1000903"][0].getValue().toString().toInt());
+        ionstring += "z";
+        pos = ion.getMonoWeight(Residue::ZIon, tr.getProduct().getChargeState()) / (double) tr.getProduct().getChargeState();
+      }
+      else if (interpretation.hasCVTerm("MS:1001229") && interpretation.hasCVTerm("MS:1000903"))
+      {
+        ion = sequence.getPrefix(interpretation.getCVTerms()["MS:1000903"][0].getValue().toString().toInt());
+        ionstring += "a";
+        pos = ion.getMonoWeight(Residue::AIon, tr.getProduct().getChargeState()) / (double) tr.getProduct().getChargeState();
+      }
+      else if (interpretation.hasCVTerm("MS:1001224") && interpretation.hasCVTerm("MS:1000903"))
+      {
+        ion = sequence.getPrefix(interpretation.getCVTerms()["MS:1000903"][0].getValue().toString().toInt());
+        ionstring += "b";
+        pos = ion.getMonoWeight(Residue::BIon, tr.getProduct().getChargeState()) / (double) tr.getProduct().getChargeState();
+      }
+      else if (interpretation.hasCVTerm("MS:1001231") && interpretation.hasCVTerm("MS:1000903"))
+      {
+        ion = sequence.getPrefix(interpretation.getCVTerms()["MS:1000903"][0].getValue().toString().toInt());
+        ionstring += "c";
+        pos = ion.getMonoWeight(Residue::CIon, tr.getProduct().getChargeState()) / (double) tr.getProduct().getChargeState();
+      }
+      else
+      {
+        unannotated = true;
+      }
+
+      if (std::find(fragment_types.begin(), fragment_types.end(), ionstring) == fragment_types.end())
+      {
+        unannotated = true;
+      }
+
+      if (interpretation.hasCVTerm("MS:1000903"))
+      {
+        // product ion series ordinal
+        ionstring += interpretation.getCVTerms()["MS:1000903"][0].getValue().toString();
+      }
+      else
+      {
+        unannotated = true;
+      }
+
+      if (interpretation.hasCVTerm("MS:1001524") && enable_losses) // fragment ion neutral loss
+      {
+        double nl = interpretation.getCVTerms()["MS:1001524"][0].getValue().toString().toDouble();
+        if (nl == -18)
+        {
+          ionstring += "-18";
+          static const EmpiricalFormula neutralloss_h2o("H2O"); // -18 H2O loss
```

Comment:
maybe move all `static const` stuff together to the start of the function? makes it maybe more readable?